jhdf/src/main/java/io/jhdf/dataset/chunked/ChunkedDatasetBase.java

```diff
@@ -322,10 +319,136 @@ public long getStorageInBytes() {
 	}
 
 	@Override
-	public ByteBuffer getSliceDataBuffer(long[] offset, int[] shape) {
-		throw new UnsupportedHdfException("Chunked datasets don't support slice reading");
+	public ByteBuffer getSliceDataBuffer(long[] sliceOffset, int[] sliceShape) {
+		logger.trace("Getting slice data buffer for {} with shape {}", Arrays.toString(sliceOffset), Arrays.toString(sliceShape));
+
+		final int elementSize = getDataType().getSize();
+		final int sliceSize = Arrays.stream(sliceShape).reduce(1, Math::multiplyExact) * elementSize;
+		final byte[] dataArray = new byte[sliceSize];
+
+		final List<Chunk> overlappingChunks = findOverlappingChunks(sliceOffset, sliceShape);
+		for (Chunk chunk : overlappingChunks) {
+			copyChunkSliceToBuffer(chunk, sliceOffset, sliceShape, dataArray, elementSize);
+		}
+
+		return ByteBuffer.wrap(dataArray);
+	}
+
+	private List<Chunk> findOverlappingChunks(long[] offset, int[] shape) {
+		int[] chunkDims = getChunkDimensions();
+		List<Chunk> overlapping = new ArrayList<>();
+
+		int[] startChunk = new int[offset.length];
+		int[] endChunk = new int[offset.length];
+
+		for (int i = 0; i < offset.length; i++) {
+			startChunk[i] = (int)(offset[i] / chunkDims[i]);
+			endChunk[i] = (int)((offset[i] + shape[i] - 1) / chunkDims[i]);
+		}
+
+		for (int[] chunkCoords : getChunkCoordinateRange(startChunk, endChunk)) {
+			Chunk chunk = getChunk(new ChunkOffset(chunkCoords));
+			if (chunk != null) {
+				overlapping.add(chunk);
+			}
+		}
+		return overlapping;
+	}
+
+	private List<int[]> getChunkCoordinateRange(int[] start, int[] end) {
+		List<int[]> coords = new ArrayList<>();
+		int dims = start.length;
+		int[] current = Arrays.copyOf(start, dims);
+
+		while (true) {
+			coords.add(Arrays.copyOf(current, dims));
+			int dim = dims - 1;
+			while (dim >= 0) {
+				current[dim]++;
+				if (current[dim] > end[dim]) {
+					current[dim] = start[dim];
+					dim--;
+				} else {
+					break;
+				}
+			}
+			if (dim < 0) break;
+		}
+		return coords;
 	}
 
+	private void copyChunkSliceToBuffer(Chunk chunk, long[] sliceOffset, int[] sliceShape, byte[] dataArray, int elementSize) {
+		byte[] chunkData = decompressChunk(chunk);
+		int[] chunkOffset = chunk.getChunkOffset();
+		int[] chunkDims = getChunkDimensions();
+		int[] fullDims = getDimensions();
+		int rank = fullDims.length;
+
+		// Determine intersection between chunk and requested slice
+		int[] intersectStart = new int[rank];
+		int[] intersectEnd = new int[rank];
+		for (int i = 0; i < rank; i++) {
+			intersectStart[i] = Math.max((int)sliceOffset[i], chunkOffset[i]);
+			intersectEnd[i] = Math.min((int)(sliceOffset[i] + sliceShape[i]), chunkOffset[i] + chunkDims[i]);
+		}
+
+		// Skip if no intersection
+		for (int i = 0; i < rank; i++) {
+			if (intersectStart[i] >= intersectEnd[i]) return;
```

Comment:
should this ever happen if the `findOverlappingChunks` method worked correctly? Maybe throw an exception?